Homework/Marianna/AbstractionInterfaces/2_3.LightSources/Room.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace _2_3.LightSources
+{
+    class Room
+    {
+        public Room(List<ILightSource> arrayOfLightSources, double maxArea, double maxUsageOfPower)
+        {
+            Sources = arrayOfLightSources;
+            MaxAreaSize = maxArea;
+            MaxPowerUsage = maxUsageOfPower;
+        }
+
+        public List<ILightSource> Sources { get; set; }
+        public double MaxAreaSize { get; set; }
+        public double MaxPowerUsage { get; set; }
+
+        public void CheckParameters(List<ILightSource> arrayOfLightSources)
+        {
+            double sumArea = 0;
+            double sumPower = 0;
+            IEnumerable<ILightSource> query = arrayOfLightSources.OrderBy(program => program.AreaSize);
+            foreach (ILightSource program in query)
+            {
+                sumArea += program.AreaSize;
+            }
+            Console.WriteLine("Sum of Illumination area size of all light sources present in the room: {0}. ", sumArea);
+
+            IEnumerable<ILightSource> query1 = arrayOfLightSources.OrderBy(program1 => program1.PowerUsage);
```

Comment:
Do you need to sort them?
`query`, `query1`... So you have a long method with several queries, you'll have something like `query100500`?
Either name them properly or inline them like so:
```
foreach (var item in arrayOfLightSources.OrderBy(program1 => program1.PowerUsage))
```